azkaban-common/src/main/java/azkaban/sla/SlaOption.java

```diff
@@ -17,191 +17,266 @@
 package azkaban.sla;
 
 import azkaban.executor.ExecutableFlow;
-import org.joda.time.DateTime;
-import org.joda.time.format.DateTimeFormat;
-import org.joda.time.format.DateTimeFormatter;
-
+import azkaban.sla.SlaType.ComponentType;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.HashMap;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 import java.util.stream.Collectors;
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
 
+/**
+ * SLA option, which can be associated with a flow or job.
+ */
 public class SlaOption {
-
-  public static final String TYPE_FLOW_FINISH = "FlowFinish";
-  public static final String TYPE_FLOW_SUCCEED = "FlowSucceed";
-
-  public static final String TYPE_JOB_FINISH = "JobFinish";
-  public static final String TYPE_JOB_SUCCEED = "JobSucceed";
-
-  public static final String INFO_DURATION = "Duration";
-  public static final String INFO_FLOW_NAME = "FlowName";
-  public static final String INFO_JOB_NAME = "JobName";
-  public static final String INFO_EMAIL_LIST = "EmailList";
-
-  // always alert
-  public static final String ALERT_TYPE = "SlaAlertType";
+  public static final String ALERT_TYPE_EMAIL = "email";
   public static final String ACTION_CANCEL_FLOW = "SlaCancelFlow";
   public static final String ACTION_ALERT = "SlaAlert";
   public static final String ACTION_KILL_JOB = "SlaKillJob";
+
+  public static final String WEB_ID = "id";
+  public static final String WEB_DURATION = "duration";
+  public static final String WEB_STATUS = "rule";
+  public static final String WEB_ACTIONS = "actions";
+  public static final String WEB_ACTION_EMAIL = "EMAIL";
+  public static final String WEB_ACTION_KILL = "KILL";
+
   private static final DateTimeFormatter fmt = DateTimeFormat
       .forPattern("MM/dd, YYYY HH:mm");
 
-  private String type;
-  private Map<String, Object> info;
-  private List<String> actions;
+  final private SlaType type;
+  final private String flowName;
+  final private String jobName;
+  final private String duration;
+  final private boolean alert;
```

Comment:
do we need this flag as this can be inferred from whether alerting email is empty or not.